addons/point_of_sale/models/pos_session.py

```diff
@@ -388,6 +388,9 @@ def _load_account_fiscal_position(self, response, fiscal_positions):
 
     def _load_account_tax(self, response, taxes):
         response['custom']['account.tax'] = {values['id']: values for values in taxes._convert_to_dict_for_taxes_computation()}
+    def _get_available_categories(self):
+        allowed_category_ids = self.config_id.iface_available_categ_ids._get_descendants().ids
+        return self.env['pos.category'].browse(allowed_category_ids)
```

Comment:
you can just return the ids, because that's what you need anyways in the places where you call the method
```suggestion
        return self.config_id.iface_available_categ_ids._get_descendants().ids
```